### PR TITLE
[Core] Translatable entities should not rely on current channel's locales

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php [hunk 1]
```diff
@@ -38,28 +39,20 @@ class ORMTranslatableListener implements EventSubscriber
...
      * @param RegistryInterface $resourceMetadataRegistry
-     * @param LocaleContextInterface $localeContext
-     * @param LocaleProviderInterface $localeProvider
+     * @param ContainerInterface $container
      */
     public function __construct(
         RegistryInterface $resourceMetadataRegistry,
-        LocaleContextInterface $localeContext,
-        LocaleProviderInterface $localeProvider
+        ContainerInterface $container
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php]: It hurts.

File: src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php [hunk 2]
```diff
@@ -107,12 +100,18 @@ public function postLoad(LifecycleEventArgs $args)
             return;
         }
 
+        /** @var LocaleContextInterface $localeContext */
+        $localeContext = $this->container->get('sylius_resource.translation.locale_context');
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php]: It hurts bad.

File: src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php [hunk 3]
```diff
@@ -107,12 +100,18 @@ public function postLoad(LifecycleEventArgs $args)
...
+
+        /** @var LocaleProviderInterface $localeProvider */
+        $localeProvider = $this->container->get('sylius_resource.translation.locale_provider');
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ORMTranslatableListener.php]: It hurts really bad.

File: src/Sylius/Bundle/ResourceBundle/Resources/config/translation.xml [hunk 4]
```diff
@@ -26,8 +26,7 @@
     <services>
         <service id="sylius.translation.translatable_listener.doctrine.orm" class="%sylius.translation.translatable_listener.doctrine.orm.class%">
             <argument type="service" id="sylius.resource_registry" />
-            <argument type="service" id="sylius_resource.translation.locale_context" />
-            <argument type="service" id="sylius_resource.translation.locale_provider" />
+            <argument type="service" id="service_container" />
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/translation.xml]: But I have to.

Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/translation.xml]: oh-god-why

File: features/locale/channel_independent_translatable_entities_locales.feature [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        When I want to create a new translatable entity
+        Then I should be able to translate it in "English (United States)"
+        And I should be able to translate it in "German (Germany)"
+
+    @ui
+    Scenario: Using all locales to translate an entity
+        Given the store has disabled locale "Polish (Poland)"
+        When I want to create a new translatable entity
+        Then I should be able to translate it in "English (United States)"
+        But I should not be able to translate it in "Polish (Poland)"
```
Comment (michalmarcinkowski) [inline features/locale/channel_independent_translatable_entities_locales.feature]: IMO we should be able to enter translation even for disabled locales in the admin. The enable/disable feature is only for shop. Am I right @pjedrzejewski?

Comment (pjedrzejewski) [inline features/locale/channel_independent_translatable_entities_locales.feature]: Yes, otherwise things get a bit complicated. :)

**Pull request merged**